pkg/virt-handler/isolation/detector.go

```diff
@@ -159,6 +159,47 @@ func (s *socketBasedIsolationDetector) AdjustResources(vm *v1.VirtualMachineInst
 	return nil
 }
 
+const qemuProcessExecutable = "qemu-kvm"
+
+// AdjustQemuProcessMemoryLimits adjusts QEMU process MEMLOCK rlimits that runs inside
+// virt-launcher pod on the given VMI according to its spec.
+// Only VMI's with VFIO devices (e.g: SRIOV, GPU) require QEMU process MEMLOCK adjustment.
+func AdjustQemuProcessMemoryLimits(podIsoDetector PodIsolationDetector, vmi *v1.VirtualMachineInstance) error {
+	if !util.IsVFIOVMI(vmi) {
+		return nil
+	}
+
+	isolationResult, err := podIsoDetector.Detect(vmi)
+	if err != nil {
+		return err
+	}
+
+	processes, err := ps.Processes()
+	if err != nil {
+		return fmt.Errorf("failed to get all processes: %v", err)
+	}
+	qemuProcess := findIsolatedQemuProcess(processes, isolationResult.PPid())
```

Comment:
Please return a relevant error if `qemuProcess` is `nil`.

Reply:
Done